static/solo/js/menu.js

```diff
@@ -0,0 +1,169 @@
+            return (typeof v !== "undefined" && v !== null) ? v : "";
+        });
+    },
+
+    trimText = function(text, limit) {
+      var t = text || '';
+      var l = limit ? Number(limit) : 0;
+      if (text.length > l) {
+        t = t.substring(0, Number(l)) + '...';
+      }
+      return t;
+    },
+
+    assety = function(data, options){
+      var assets = [];
+      for (var i = 0; i < data.length; i++) {
+        var video = data[i];
+        
+        if (isEmpty(video)) break;
+
+        var asset = video.asset;
+        asset.assetId = video.id;
+        asset.assetTitle = video.title;
+        asset.loginId = video.loginId;
+
+        if (isset(video,'events') && video.events.length) {
+          asset.analyticsLogUrl = video.analytics;
+          asset.analyticsObj = video.events[1].data;
+        } else {
+          asset.analyticsObj = {
+            pg: isset(video,'parentId') ? video.parentId : ( isset(options,'channel') ? options.channel.id : 0 ),
+            vd: video.id, 
+            li: video.loginId
+          };
+        }
+
+        if (!asset.sources) asset.sources = [{ file: asset.videoId }];
+        asset.type = asset.type || 'youtube';
+        assets.push(asset); 
+      }
+      return assets;
+    };
+
+
+  function Menu(player, settings) {
+
+    var that = this;
+    this.allVideos = [];
+
+    this.render = function() {
+      var playlist = settings.data || [];
+      if (playlist.length < 1) return;
+
+      that.menuFrag = document.createDocumentFragment();
+      that.slideMenu = document.createElement('div');
+      that.slideMenu.setAttribute('id', 'tvp-slide-menu');
+      that.slideMenu.innerHTML = menuTemplate;
+      that.hiddenMenu = that.slideMenu.querySelectorAll('#tvp-hidden-menu')[0];
+      that.hamburguer = that.slideMenu.querySelectorAll('#tvp-hamburger-container')[0];
+      that.menuFrag.appendChild(that.slideMenu);
+
+      for (var i = 0; i < playlist.length; i++) {
+        var menuItem = playlist[i];
+        that.allVideos.push(menuItem);
+        menuItem.title = trimText(menuItem.title, 50);
+
+        var menuItemElFrag = document.createDocumentFragment();
+        var menuItemEl = document.createElement('div');
+        menuItemEl.classList.add('tvp-slide-menu-video');
+        menuItemEl.innerHTML = tmpl(itemTemplate, menuItem);
+        menuItemElFrag.appendChild(menuItemEl);
+        that.hiddenMenu.appendChild(menuItemElFrag);
+      }
+      document.body.appendChild(that.menuFrag);
+      SimpleScrollbar.initEl(that.hiddenMenu);
+      that.bindMenuEvent();
+      that.bindClickEvent();
+    };
+
+    this.bindMenuEvent = function() {
+      var toggles = document.querySelectorAll('.tvp-hamburger');
+      for (var i = toggles.length - 1; i >= 0; i--) {
+        toggles[i].onclick = function() {
+            that.toggleMenu();
+        };
+      }
+    };
+
+    this.bindClickEvent = function(){
```

Comment:
Lets take a look to delegate this click handling to a single element, touch base with me, its long to explain.